src/coreComponents/common/MpiWrapper.hpp

```diff
@@ -1115,6 +1125,30 @@ void MpiWrapper::reduce( Span< T const > const src, Span< T > const dst, Reducti
   reduce( src.data(), dst.data(), LvArray::integerConversion< int >( src.size() ), getMpiOp( op ), root, comm );
 }
 
+// Mpi helper function to return  tuple containing the max value of element 0 in the tuple if rank 0
+// For rank > 0 input tuple is returned
+template< typename tuple_type >
+tuple_type
+MpiWrapper::tupleMaxLoc( tuple_type tuple, MPI_Comm comm )
+{
+  // recieve buffer
+  std::vector< tuple_type > l_rcv_buffer;
+
+  int const myRank  = MpiWrapper::commRank( comm );
+  int const numProcs = MpiWrapper::commSize( comm );
+  if( myRank == 0 )
+  {
+    l_rcv_buffer.resize( numProcs );
+  }
+  MPI_Gather( &tuple, sizeof(tuple_type), MPI_BYTE, &l_rcv_buffer[0], sizeof(tuple_type), MPI_BYTE, 0, comm );
```

Comment:
First, I would suggest only passing the buffer to rank 0 with something like:
```suggestion
  MPI_Gather( &tuple, sizeof(tuple_type), MPI_BYTE, myRank == 0 ? l_rcv_buffer.data() : nullptr, sizeof(tuple_type), MPI_BYTE, 0, comm );
```
second,

I think this is actually kind of problematic. I don't think a tuple is always stored as a contiguous block of memory which is what `MPI_Gather` expects. I believe that this can lead to weird behaviors if the data is not contiguous. You could copy it into a buffer with contiguous memory and use that to perform the gather but it is somewhat involved. Do we really need this max over tuples? 

@wrtobin and @rrsettgast can probably comment more on this.

Reply:
Also, this really expects a pair not a tuple and I think we should have some checks on the types contained in the tuple. I.e. The first type must be comparable.

Reply:
It is not guaranteed by the standard. The tuple should be explicitly packed/unpacked.

Reply:
Yeah the types contained within the tuple would all need to be standard layout IIRC for this to be valid. Packing/unpacking is the easier lift than enforcing it through type trait checks.

Reply:
submitted change to use struct for storage and serialization to/from byte string  for mpiallgather